concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -365,13 +380,22 @@ extern "C" {
         appdata_dir_len: i64,
         runner_ptr_ptr: *mut *mut consensus_runner,
     ) -> i64;
-    #[allow(improper_ctypes)]
     pub fn startBaker(consensus: *mut consensus_runner);
     pub fn receiveBlock(
         consensus: *mut consensus_runner,
         genesis_index: u32,
         block_data: *const u8,
-        data_length: i64,
+        data_length: u64,
+        // A mutable *pointer that will be written to
+        // by the consensus layer with continuation required for
+        // executing the just received block.
+        execute_block_ptr_ptr: *mut *mut execute_block,
+    ) -> i64;
+    pub fn executeBlock(
+        consensus: *mut consensus_runner,
+        // Pointer to the continuation for
+        // adding the block to the tree.
+        execute_block_callback: *mut execute_block,
```

Comment:
If we have the callback struct we should use it consistently. Though I recommend not having it.

Reply:
This now takes a 'ExecuteBlockCallback'. 
I had to add `#[allow(dead_code)]` to the `callback` field of the type to avoid warnings. I don't know if there is a better way? 
@abizjak @td202

Reply:
I would recommend being consistent with other such definitions and not have an irrelevant type wrapper.

I don't having this struct wrapper improves anything.

Reply:
`ExecuteBlockCallback` is now defined as `pub struct ExecuteBlockCallback(*mut execute_block);` avoiding the `allow(dead_code)` chunk. 

I have moreover documented on the `ExecuteBlockCallback` why the opaque pointer is kept in an owned type.